nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/test/java/org/apache/nifi/processors/kite/TestCSVToAvroProcessor.java

```diff
@@ -191,57 +192,40 @@ public void testEmptyContent() throws IOException {
-        Assert.assertEquals("Escape should match",
-                "\\", processor.props.escape);
-        Assert.assertEquals("Header flag should match",
-                false, processor.props.useHeader);
-        Assert.assertEquals("Lines to skip should match",
-                0, processor.props.linesToSkip);
+        runner.assertNotValid();
+        runner.setProperty(ConvertCSVToAvro.SCHEMA, SCHEMA.toString());
+        runner.assertValid();
 
-        runner.setProperty(ConvertCSVToAvro.CHARSET, "utf16");
-        runner.setProperty(ConvertCSVToAvro.DELIMITER, "|");
-        runner.setProperty(ConvertCSVToAvro.QUOTE, "'");
-        runner.setProperty(ConvertCSVToAvro.ESCAPE, "\u2603");
-        runner.setProperty(ConvertCSVToAvro.HAS_HEADER, "true");
-        runner.setProperty(ConvertCSVToAvro.LINES_TO_SKIP, "2");
-
-        // check updates
-        processor.createCSVProperties(context);
-        Assert.assertEquals("Charset should match",
-                "utf16", processor.props.charset);
-        Assert.assertEquals("Delimiter should match",
-                "|", processor.props.delimiter);
-        Assert.assertEquals("Quote should match",
-                "'", processor.props.quote);
-        Assert.assertEquals("Escape should match",
-                "\u2603", processor.props.escape);
-        Assert.assertEquals("Header flag should match",
-                true, processor.props.useHeader);
-        Assert.assertEquals("Lines to skip should match",
-                2, processor.props.linesToSkip);
+        runner.enqueue(streamFor("1,green\n2,blue,\n3,grey,12.95"));
+        runner.run();
+
+        long converted = runner.getCounterValue("Converted records");
+        long errors = runner.getCounterValue("Conversion errors");
+        Assert.assertEquals("Should convert 3 rows", 3, converted);
+        Assert.assertEquals("Should reject 0 row", 0, errors);
+
+        runner.assertTransferCount("success", 1);
+        runner.assertTransferCount("failure", 0);
+        runner.assertTransferCount("incompatible", 0);
     }
 
     @Test
-    public void testBasicConversionNoErrors() throws IOException {
+    public void testExpresionLanguageBasedCSVProperties() throws IOException {
         TestRunner runner = TestRunners.newTestRunner(ConvertCSVToAvro.class);
         runner.assertNotValid();
         runner.setProperty(ConvertCSVToAvro.SCHEMA, SCHEMA.toString());
         runner.assertValid();
 
-        runner.enqueue(streamFor("1,green\n2,blue,\n3,grey,12.95"));
+        runner.setProperty(ConvertCSVToAvro.DELIMITER, "${csv.delimiter}");
+        runner.setProperty(ConvertCSVToAvro.QUOTE, "${csv.quote}");
+
+        HashMap<String, String> flowFileAttributes = new HashMap<String,String>();
+        flowFileAttributes.put("csv.delimiter", "|");
+        flowFileAttributes.put("csv.quote", "~");
+
+        runner.enqueue(streamFor("1|green\n2|~blue|field~|\n3|grey|12.95"), flowFileAttributes);
```

Comment:
Might be helpful to check that the fields are converted as expected (by reading the resulting Avro file(s)) instead of just the total number of records produced.